src/zcash/JoinSplit.cpp

```diff
@@ -151,6 +144,7 @@ class JoinSplitCircuit : public JoinSplit<NumInputs, NumOutputs> {
 
     bool verify(
         const ZCProof& proof,
+        ProofVerifier& verifier,
```

Comment:
@ebfull explained in our pairing that we don't want this to be `const` because `ProofVerifier::check()` is not in fact specified as `const`, precisely because a future `ProofVerifier::Batch()` *will* require mutability.